azkaban-common/src/main/java/azkaban/executor/ExecutorHealthChecker.java

```diff
@@ -97,12 +97,7 @@ public void checkExecutorHealth() {
       final Optional<Executor> executorOption = entry.getKey();
       if (!executorOption.isPresent()) {
         final String finalizeReason = "Executor id of this execution doesn't exist.";
-        for (final ExecutableFlow flow : entry.getValue()) {
-          logger.warn(
-              String.format("Finalizing execution %s, %s", flow.getExecutionId(), finalizeReason));
-          ExecutionControllerUtils
-              .finalizeFlow(this.executorLoader, this.alerterHolder, flow, finalizeReason, null);
-        }
+        finalizeFlows(entry.getValue(), finalizeReason);
```

Comment:
I feel the function renaming is ling overdue:

finalizeFlowsIfExecutorIsUnhealthy() will better reflect what the function does.

Also - midway in this function there is a ToDo from Jamie requesting a metric for Http response times? Do you know what is that for?

Reply:
The private method `finalizeFlows` was added by me as part of this PR as a convenience wrapper over `ExecutionControllerUtils.finalizeFlows`. 
The purpose of the method is specifically to finalize any flows with the given reason. It does not take into account whether there was an error specific to the executor ( a higher level method will make that judgement) 
So I would prefer to keep the method name unchanged. Let me know your thoughts.

Reply:
The todo from Jamie seems to be talking about measuring the response time for the healthcheck. That would be valuable, unless we are already capturing executor response latencies as some other metric.

Reply:
I had meant renaming the function checkExecutorHealth(). 
Two suggestions being 
- finalizeFlowsIfExecutorIsUnhealthy()
- finalizeFlowsForUnhealthyExecutor()

Reply:
Ah, got it now. However I feel the current name of the method is still more reflective of what it's intended to do. For instance something like `finalizeFlows..` does immediately reflect that an explicit check (in this case an ajax api call) is included as part of the method.

Reply:
Doesn't finalizeFlowsIfExecutorIsUnhealthy() not tell the flows will be finalized if the Executor is unhealthy?

Reply:
Decided to keep the current name for now. 
`finalizeFlows...` does not capture the checking and is more reflective of 'how' rather than 'what' the function is doing.  Think about what happens if/when the function starts doing more than just finalizing flows.
We could of course choose something like `checkExecutorHealthAndRemediateIfNeeded` :) but that seems long and unnecessary when a short comment in the function description will suffice.